test/com/google/javascript/jscomp/LiveVariablesAnalysisTest.java

```diff
@@ -659,12 +659,29 @@ private static void assertEscaped(String src, String name) {
   }
 
   private static void assertNotEscaped(String src, String name) {
-    for (Var var : computeLiveness(src, false).getEscapedLocals()) {
+    assertNotEscaped(src, name, true);
+  }
+
+  /**
+   * @param wrapInFunction Whether the input should be wrapped in a function.
+   */
+  private static void assertNotEscaped(String src, String name, boolean wrapInFunction) {
+    for (Var var : computeLiveness(src, false, wrapInFunction).getEscapedLocals()) {
       assertThat(var.getName()).isNotEqualTo(name);
     }
   }
 
   private static LiveVariablesAnalysis computeLiveness(String src, boolean async) {
+    return computeLiveness(src, async, true);
+  }
+
+  /**
+   * @param wrapInFunction Whether the input should be wrapped in a function e.g. <code>var a = 1;
+   *     </code> -> <code>function f() { var a = 1; }</code>
+   * @param async Whether the wrapper function should be {@code async}.
+   */
+  private static LiveVariablesAnalysis computeLiveness(
+      String src, boolean async, boolean wrapInFunction) {
```

Comment:
Could we replace the two boolean parameters with an enum? something like

``` java
enum Wrapper {
  NONE,
  FUNCTION,
  ASYNC_FUNCTION
}
```

Reply:
Good idea, the enum makes more sense. Done